### DISCUSSION TITLE
Slow plotting...

### DISCUSSION THREAD

**Discussion by lokiby** [madMAx43v3r/chia-plotter#224] [Q&A]
Hello everyone. I apologize for the English - I am writing via google translator.
Downloaded version madMAx43v3r_chia-plotter_win_build_v0.0.3-alpha
Tried on virtual machines:
1) ESXi - Win 7 x64 (32 CPU (Intel Silver 4210) / 128 RAMM / 1.9TV SSD RAID 0 / 3.5 TV HDD RAID 5) - made a 110GB disk from RAM (for temp2), temp1 - ssd, final folder - hdd
2) Proxmox - Win 7 x64 (12 CPU / 48 RAMM / 880GB SSD NVMe RAID 0 / 1.6TB HDD) - made temp2 - ssd, temp1 - ssd, final folder - hdd
3) physical machine - AMD FX 8320 (@ 4.4Ghz) / 32 RAMM / 1TB SSD / 1TB HDD - made temp2 - ssd, temp1 - ssd, final folder - hdd
Launched it on the first machine (ESXi) with the following parameters:
chia_plot.exe -p pulic_key -f farmer_key -t SSD -2 RAM -d HDD -r 20 -u 7 -n 1
pause

On the rest with changed folders and fewer threads.
The problem is that the results are low everywhere:
P1 Table 1 - 100-120 sec
P1 Table 2 - 1100-1300 sec
Then I realized that there was no point.
The problem is on completely different machines, so I'm probably doing something wrong?

**Comment by farchord** [comment]
Just a thought, have you tried to rule out ESXi as a culprit? Make a Live Linux USB drive and try to do a single plot. If it runs fine, ESXi is probably to blame (Or a setting on it).

**Reply by lokiby** [reply]
> Just a thought, have you tried to rule out ESXi as a culprit? Make a Live Linux USB drive and try to do a single plot. If it runs fine, ESXi is probably to blame (Or a setting on it).

I wrote that I tried it on 2 different virtual machines and on one physical machine. Read carefully.
Runs slowly everywhere, on all 3 systems.

**Reply by Kiggles** [reply]
128GB system memory with 110 _**GiB**_ ram drive will not leave enough memory for more than 16 threads, and likely only on headless linux servers, not a full Windows desktop environment.

First, make sure your ram disk is 110 **_GiB_** not 110GB, next, attempt the same but -r 8 , or even -r 4 to maybe guarantee you're not just dipping into swap file becoming the bottleneck.

**Reply by lokiby** [reply]
> 128GB system memory with 110 _**GiB**_ ram drive will not leave enough memory for more than 16 threads, and likely only on headless linux servers, not a full Windows desktop environment.
> 
> First, make sure your ram disk is 110 **_GiB_** not 110GB, next, attempt the same but -r 8 , or even -r 4 to maybe guarantee you're not just dipping into swap file becoming the bottleneck.

On other systems without a RAM disk, but with an SSD, the same problems. He does it very slowly.
On the AMD system I put -r 4 and still slow, and there is 1TB SSD and 32 GB RAMM

**Reply by Kiggles** [reply]
I am disregarding the other systems. Use fewer threads as a test on the first Intel system. There is a common problem, but focus on a single platform now to isolate it.

But if you want to do everything at once, the only commonality is Win7. So, start isolating variables.
